rpc/src/rpc.rs

```diff
@@ -508,30 +610,38 @@ impl JsonRpcRequestProcessor {
         let encoding = encoding.unwrap_or(UiAccountEncoding::Binary);
         optimize_filters(&mut filters);
         let keyed_accounts = {
-            if let Some(owner) = get_spl_token_owner_filter(program_id, &filters) {
+            if let Some(owner) = get_spl_token_owner_filter(&program_id, &filters) {
                 self.get_filtered_spl_token_accounts_by_owner(
-                    &bank,
+                    Arc::clone(&bank),
                     program_id,
-                    &owner,
+                    owner,
                     filters,
                     sort_results,
-                )?
-            } else if let Some(mint) = get_spl_token_mint_filter(program_id, &filters) {
+                )
+                .await?
+            } else if let Some(mint) = get_spl_token_mint_filter(&program_id, &filters) {
                 self.get_filtered_spl_token_accounts_by_mint(
-                    &bank,
+                    Arc::clone(&bank),
                     program_id,
-                    &mint,
+                    mint,
                     filters,
                     sort_results,
-                )?
+                )
+                .await?
             } else {
-                self.get_filtered_program_accounts(&bank, program_id, filters, sort_results)?
+                self.get_filtered_program_accounts(
+                    Arc::clone(&bank),
+                    program_id,
+                    filters,
+                    sort_results,
+                )
+                .await?
             }
         };
-        let accounts = if is_known_spl_token_id(program_id)
+        let accounts = if is_known_spl_token_id(&program_id)
```

Comment:
does getProgramAccounts actually return _program_ data? If so, I suspect we
should do the encoding in the bg pool (programs are large)

Reply:
Not usually. It returns accounts owned by the specified program. If the specified program in question was ‘the loader program’ then yeah, you'd get program data, but most RPC would fatal with a `scan aborted: The accumulated scan results exceeded the limit` before that happened.